src/main/scala/uk/gov/nationalarchives/omega/jms/JmsRRClient.scala

```diff
@@ -0,0 +1,222 @@
+
+import java.util.UUID
+import java.util.concurrent.ConcurrentHashMap
+import scala.concurrent.duration.DurationInt
+import scala.util.{Failure, Success}
+
+case class RequestMessage(body: String, sid: String)
+case class ReplyMessage(body: String)
+
+/**
+ * A JMS Request-Reply client.
+ * Suitable for being packaged as a library for abstracting
+ * away the complexities of JMS and jms4s.
+ *
+ * @param requestMap - a map of the requests sent and their reply handlers
+ * @param consumer - unused
+ * @param producer - the JmsProducer used to send the message
+ * @tparam F
+ */
+class JmsRRClient[F[_]: Async: Logger](requestMap: ConcurrentHashMap[String, ReplyMessageHandler[F]])(consumer: Unit, producer: JmsProducer[F]) {
+
+  /**
+   *
+   * @param requestQueue - the queue to send the message to
+   * @param jmsRequest  - the request message
+   * @param replyMessageHandler - the reply handler
+   * @return
+   */
+  def request(requestQueue: String, jmsRequest: RequestMessage, replyMessageHandler: ReplyMessageHandler[F]): F[Unit] = {
+
+    /* calls the JmsProducer send method and returns an optional message ID */
+    val sender: F[Option[String]] = producer.send { mf =>
+      val jmsMessage: F[JmsMessage.JmsTextMessage] = mf.makeTextMessage(jmsRequest.body)
+      Async[F].map(jmsMessage)(jmsMessage =>
+        jmsMessage.setStringProperty("sid", jmsRequest.sid) match {
+          case Success(_) => (jmsMessage, QueueName(requestQueue))
+          case Failure(e) =>
+            // TODO (RW) log the error and find a way out of sending the message
+            (jmsMessage, QueueName(requestQueue))
+        })
+    }
+
+    Async[F].flatMap(sender) {
+      case Some(messageId) =>
+        Async[F].delay {
+          val _ = requestMap.put(messageId, replyMessageHandler)
+          ()
+        }
+      case None =>
+        Async[F].raiseError(new IllegalStateException("No messageId obtainable from JMS but application requires messageId support"))
+    }
+  }
+}
+
+object JmsRRClient {
+
+  type ReplyMessageHandler[F[_]] = ReplyMessage => F[Unit]
+
+  private val defaultConsumerConcurrencyLevel = 1 // 10
+  private val defaultConsumerPollingInterval = 50.millis
+  private val defaultProducerConcurrencyLevel = 1 // 10
+
+  /**
+   * Create a JMS Request-Reply Client for use with Apache Active MQ.
+   *
+   * @param endpoint the ActiveMQ broker endpoint to connect to.
+   * @param credentials the credentials for connecting to the ActiveMQ broker.
+   * @param customClientId an optional Custom client ID to identify this client.
+   *
+   * @param replyQueue the queue that replies should be consumed from.
+   *
+   * @return The resource for the JMS Request-Reply Client.
+   */
+  def createForActiveMq[F[_]: Async: Logger](endpoint: HostBrokerEndpoint, credentials: UsernamePasswordCredentials, customClientId: Option[F[String]] = None)(replyQueue: String): Resource[F, JmsRRClient[F]] = {
```

Comment:
Given that we don't currently need support for ActiveMQ, could we remove it? 

Or perhaps include it for this ticket, but remove it in a subsequent one, so we can easily revert?

Reply:
I've decided to remove ElasticMQ support completely for now. It will be easy enough to add again. The fact is that the tests were timing out and I don't know why but I don't think we should be spending a lot of time on it now.

Reply:
I assume you mean ActiveMQ?

Good.

Reply:
Edited my comment.